std/array.d

```diff
@@ -2819,36 +2819,114 @@ if (isDynamicArray!A)
 
     /**
      * Appends `item` to the managed array.
+     * Params:
+     *  item = an element that can be placed in the array,
+     *         or a range of such elements.
      */
-    void put(U)(U item) if (canPutItem!U)
```

Comment:
If there are two overloads, one with a constraint of `if (A)` and the other with `if (!A)`, then it's irrelevant to the user (being an internal implementation consideration). Phobos is full of these things, and they make understanding what's going on pointlessly difficult.

A related problem is the scattering of overloads all over the place, generally in an undocumented and arbitrary manner. `split` is a good example of that.